api/v1/sp-eligibility-criteria.go

```diff
@@ -1,41 +1,80 @@
 package api
 
 import (
+	"encoding/json"
 	"net/http"
 
+	"github.com/data-preservation-programs/spade-tenant/db"
 	"github.com/labstack/echo/v4"
+	"gorm.io/gorm"
 )
 
-type EligibilityCriteria struct {
-	Clauses []EligibilityClause `json:"clauses"`
+func (a *apiV1) ConfigureSpEligibilityCriteriaRouter(e *echo.Group) {
+	g := e.Group("/sp")
+	g.POST("/eligibility-criteria", a.handleSetSpEligibilityCriteria)
+	g.GET("/eligibility-criteria", a.handleGetSpEligibilityCriteria)
 }
 
-type EligibilityClause struct {
-	Attribute string `json:"attribute"`
-	Operator  string `json:"operator"`
-
-	Value interface{} `json:"value"` // TODO: type - either []string or string
+type TenantSPEligibilityClausesResponse struct {
+	ClauseAttribute string                `json:"attribute" gorm:"primaryKey"`
+	ClauseOperator  db.ComparisonOperator `json:"operator" gorm:"type:comparison_operator;not null"`
+	ClauseValue     string                `json:"value" gorm:"not null"`
 }
 
 // handleSetSpEligibilityCriteria godoc
 //
-//		@Summary		Set sp eligibility criteria
-//		@Security apiKey
-//	  @Param 			elibility_criteria body EligibilityCriteria true "New eligibility criteria to update to"
-//		@Produce		json
-//		@Success		200	{object}	ResponseEnvelope{response=EligibilityCriteria}
-//		@Router			/sp/eligibility-criteria [post]
-func handleSetSpEligibilityCriteria(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+//	@Summary		Set sp eligibility criteria
+//	@Security		apiKey
+//	@Param			eligibility_criteria body EligibilityCriteria true "New eligibility criteria to update to"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=EligibilityCriteria}
+//	@Router			/sp/eligibility-criteria [post]
+//
+
+func (a *apiV1) handleSetSpEligibilityCriteria(c echo.Context) error {
```

Comment:
hmm, it seems like this API basically only allows for "append only", adding new EligibilityCriteria as it iterates over all the provided ones and sets TenantID. That is okay for a POST (i.e, `CREATE`  endpoint), but I think we also need to add an UPDATE and DELETE method, so the tenant can update existing criteria and delete them if needed (i.e, when providing an existing TenantID/Attribute composite primary key)

Reply:
I can add that should be straight forward. It wasnt part of the original api which is why it is missing

Reply:
Decided to implement just delete from a discussion offline. High level is post and delete accomplishes what we want